ContactDetailsApi/V2/Infrastructure/IPWhitelistMiddleware.cs

```diff
@@ -0,0 +1,62 @@
+using ContactDetailsApi.V2.Domain;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace ContactDetailsApi.V2.Infrastructure
+{
+    public class IPWhitelistMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly ILogger<IPWhitelistMiddleware> _logger;
+        private readonly HashSet<string> _safelist;
+
+        public IPWhitelistMiddleware(
+            RequestDelegate next,
+            ILogger<IPWhitelistMiddleware> logger)
+        {
+            _next = next;
+            _logger = logger;
+            try
+            {
+                var safelist = Environment.GetEnvironmentVariable("WHITELIST_IP_ADDRESS");
+                _logger.LogInformation("whitelist ip address is {safelist}", safelist);
+                var ips = safelist.Split(';');
+                _safelist = new HashSet<string>(ips);
+
+            }
+            catch (Exception ex)
+            {
+                _logger.LogInformation($" cannot get env var {ex.Message}");
+            }
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            if (context.Request.Path == "/api/v2/servicesoft/contactDetails")
+            {
+                
+                var remoteIp = context.Connection.RemoteIpAddress;
+                _logger.LogDebug("Request from Remote IP address: {RemoteIp}", remoteIp);
+
+                if (!_safelist.Contains(remoteIp.ToString()))
```

Comment:
Are we okay with this endpoint being inaccessible from Hackney's network?
Because the way read this is: `allow only the service soft`.

In case we actually do want to have access to this endpoint, I recommend creating a wrapper endpoint:
```cs
 [HttpGet]
 [Route("hackney/contactDetails")]
 [AuthorizeEndpointByGroups("SOME_HACKNEY_REPORTING_GROUP")]
 public async Task<IActionResult> FetchAllContactDetailsByUprn()
 {
     // This is the original endpoint that anyone can access so long they have Hackney g group or bearer
     // some endpoint logic ...
 }

 [HttpGet]
 [Route("servicesoft/contactDetails")]
 [IPWhitelist("SERVICE_SOFT_IP_WHITELIST")]
 public async Task<IActionResult> FetchAllContactDetailsByUprn()
 {
     // This is a wrapper method the original endpoint that has service soft facing auth & its restrictions
     // All this method does is call the method above ^^^ and returns its results.
     return FetchAllContactDetailsByUprn();
 }
```